anx/provider/loadbalancer/address/address.go

```diff
@@ -37,8 +38,8 @@ type Manager interface {
 }
 
 // NewWithPrefixes creates a new Manager instance for a list of Prefix identifiers
-func NewWithPrefixes(ctx context.Context, apiClient api.API, legacyClient client.Client, prefixes []string) (Manager, error) {
-	m := newMgr(ctx, apiClient, legacyClient)
+func NewWithPrefixes(ctx context.Context, apiClient api.API, legacyClient client.Client, clusterName string, prefixes []string) (Manager, error) {
```

Comment:
I'd say when a prefix is configured instead of discovered, we should give a way to configure the address as well - having either "configured prefix and address" or "autodiscovered prefix and address"

and since this feature is a bit time-critical, I'd be ok with at first only supporting the second way, "autodiscovered prefix and address" and leave the "configured prefix" way at the fixed address until a followup PR

Reply:
This seems a non-issue to me, given that `ctx` and the clients are basically boilerplate.
Is this a common best practice limitation for Golang code bases?

Reply:
the problem codeclimate bugs here is a non-issue, ack - but see my other comment as well

Reply:
Github messed up the comment order here, I was only referring to the codeclimate review.

From my understanding the behavior in this PR is exactly what you describe (although implicitly) - tagged prefixes & addresses work, configured prefixes and legacy canonical IP addresses work.

Configuring addresses could of course be added later on, but AFAIK there is little to no demand for that.

Reply:
@LittleFox94 I need a little clarification: Do you want a strict separation of configured prefixes and auto discovery? As in: For configured prefixes CCM will not attempt to discover VIPs in them.
Currently CCM always looks for tagged addresses (if they are tagged, they are to be used). From my understanding some of your other comments imply that this should not happen.

Reply:
I've now separated the code paths, trying to stick to the established pattern of separate initializer functions.

Reply:
`LoadBalancerPrefixIdentifiers` being set in the config will decide whether both prefixes and addresses are autodiscovered, or prefixes are read from config and addresses are calculated.

Reply:
> As in: For configured prefixes CCM will not attempt to discover VIPs in them.

That's what I meant, yes - but not really important for this PR; the current config is a mess anyway